src/main/scala/com/scalacamp/hometasks/first/RetryUtil.scala

```diff
@@ -0,0 +1,28 @@
+package com.scalacamp.hometasks.first
+
+import scala.annotation.tailrec
+import scala.concurrent.duration.FiniteDuration
+
+
+object RetryUtil {
+  @tailrec
+  final def retry[A](block: () => A,
+                     acceptResult: A => Boolean,
+                     retries: List[FiniteDuration]): A = {
+    retries match {
+      case Nil => block()
+      case head :: tail => {
+        Thread.sleep(head.toMillis)
+        val result = block()
+        if (acceptResult(result)) {
+          result
+        } else {
+          tail match {
+            case Nil => result
+            case _ => retry(block, acceptResult, tail)
+          }
+        }
+      }
+    }
```

Comment:
what happens in case of:
```scala
retry[Int](
      () => 2,
      r => r > 2,
      List(1.seconds)
    )
```
i'm afraid no retries will be execute at all, but 1 retry is expected.

Reply:
Actually, the function is executed once as expected
I added a unit test for this case

Reply:
@FastForward77 the function should be executed twice. The second execution should be retry. And 1 second should be the time that retry function waits for between the function eecution and the retry.

Reply:
Do you mean that first execution should be without any delay?
For example for:
```
retry[Int](
      () => 2,
      r => r > 2,
      List(1.seconds, 2.seconds)
    )
```
First excecution
Delay 1 second
Second excecution
Delay 2 second
Third execution

Reply:
Exactly!

Reply:
now it's fixed